enhancements/20230327-harvester-seeder-integration.md

```diff
@@ -0,0 +1,244 @@
+
+#### Discover hardware details via Harvester
+
+### User Experience In Detail
+
+A harvester user wants to discover underlying hardware information from harvester.
+
+Currently, this is not possible, however once the seeder addon is enabled, the user can defined inventory objects for their Harvester nodes, to map individual Harvester nodes to the underlying hardware.
+
+Once the addon is enabled, and the Inventory object is defined the end user can see underlying hardware information in the Harvester UI.
+
+This will include information such as (but not limited to):
+* Hardware manufacture
+* Hardware serial number
+* Hardware model
+* Hardware events
+
+![](./20230327-harvester-seeder-integration/hardware-discovery.png)
+
+![](./20230327-harvester-seeder-integration/events.png)
+
+
+#### Power cycle nodes via Harvester
+
+### User Experience In Detail
+
+A harvester user wishes to reboot/shutdown/power-on underlying hardware from harvester.
+
+Once the seeder addon is enabled, the users can power cycle underlying nodes by annotating the node with seeder specific annotations. (these changes will be made available via UI+Harvester API)
+
+`metal.harvesterhci.io/actionRequested: poweron|shutdown|reboot` - triggers specific power action on the node.
+
+Once the action is performed seeder will set additional annotations on the node object
+
+`metal.harvesterhci.io/actionStatus: running|failed|complete` - will indicate status of power action. 
+`metal.harvesterhci.io/rufioJobName` - will indicate the underlying job name for power action.
+
+### API changes
+Similar to PCIDeviceClaims, the UI will allow end users to define a secret for bmc credentials as well as associated inventory objects.
+
+The above two changes can be handled directly by the UI.
+
+Once and inventory is associated with a node, the associated nodeObject needs to be annotated with additional information:
+
+`harvesterhci.io/inventoryName: inventoryName`
+
+There will be changes to the node api: https://github.com/harvester/harvester/blob/master/pkg/api/node/formatter.go#L40
+
+Changes will involve additional resource Actions:
+
+* if annotation `harvesterhci.io/inventoryName` is present and node is in `maintenanceMode`, but no `metal.harvesterhci.io/actionRequested` is present then the user can invoke action `powerAction` with input `action`, which will annotate the node and trigger power action on underlying inventory
+*  if annotation `harvesterhci.io/inventoryName` is present, then user can invoke action `lastPowerActionStatus` which will report the status of the last power action.
+
+
+## Design
+The seeder addon will introduce a few new CRD's, however only the following are used in embedded mode:
+
+* inventories.metal.harvesterhci.io
+* clusters.metal.harvesterhci.io
+* jobs.bmc.tinkerbell.org
+* machines.bmc.tinkerbell.org
+* tasks.bmc.tinkerbell.org
+
+### Implementation Overview
+
+seeder will run as a deployment in the `harvester-system` namespace in the cluster.
```

Comment:
Do we allow to enable the add-on during the installation stage?

Reply:
this addon is optional, and similar to the pcidevices addon, and enabled post install.